src/node_nanomsg.cc

```diff
@@ -87,6 +87,70 @@ NAN_METHOD(Connect) {
   info.GetReturnValue().Set(Nan::New<Number>(nn_connect(s, *addr)));
 }
 
+class BindThread : public Nan::AsyncWorker {
+public:
+  BindThread(Nan::Callback *callback, int s, char *addr)
+      : Nan::AsyncWorker(callback), s(s), addr(addr) {}
+  ~BindThread() {}
+
+  // Executed inside worker-thread.
+  void Execute() {
+    int bound = nn_bind(s, addr);
+  }
+  void HandleOKCallback() {
+    Nan::HandleScope scope;
+    Local<Value> argv[] = { Nan::New<Number>(bound) };
+    callback->Call(1, argv);
+  };
+
+private:
+  int s;
+  char *addr;
+  int bound;
+};
+
+NAN_METHOD(asyncBind) {
+  Nan::Callback *callback = new Nan::Callback(info[2].As<Function>());
+
+  int s = Nan::To<int>(info[0]).FromJust();
+  String::Utf8Value addr(info[1]);
+
+  Nan::AsyncQueueWorker(new BindThread(callback, s, *addr));
+  info.GetReturnValue().Set(Nan::Undefined());
+}
+
+class ConnectThread : public Nan::AsyncWorker {
+public:
+  ConnectThread(Nan::Callback *callback, int s, char *addr)
+      : Nan::AsyncWorker(callback), s(s), addr(addr) {}
+  ~ConnectThread() {}
+
+  // Executed inside worker-thread.
+  void Execute() {
+    int connected = nn_connect(s, addr);
+  }
+  void HandleOKCallback() {
+    Nan::HandleScope scope;
+    Local<Value> argv[] = { Nan::New<Number>(connected) };
```

Comment:
How is `connected` defined in Execute, but accessed in `HandleOKCallback`? wat :fire: